### PR TITLE
Replace pkg_resources

### PR DESCRIPTION
**Pull request by fschulze**
Just importing pkg_resources becomes slower and slower the more packages are installed. Even on fast machines with SSDs this can add a second or more to the startup time.

### CODE REVIEW

**Comment by fschulze** [comment]
This will most likely fail for packages where the resources have to be extracted, as the used context wrapper will cleanup upon exit and the current ``*TemplateFile`` implementations currently try to access the file later. There should most likely be a separate class like ResourceTemplate and the loader should return that instead.

**Comment by malthe** [comment]
@fschulze seems this is almost there? what remains here before we can merge

**Comment by fschulze** [comment]
See [my last comment](https://github.com/malthe/chameleon/pull/358#issuecomment-1192581061). It fails for packages which have the templates zipped. I'm also not sure whether that case is covered by tests yet (and how hard such tests would be to implement/simulate).

**Comment by malthe** [comment]
@fschulze that makes sense – perhaps we should shift this way of loading a file-based template to the `*File` classes – instead of having it in the loader?

**Comment by fschulze** [comment]
I will most likely work on that this week.

**Comment by fschulze** [comment]
The spec is now stored on the template and handled properly. The filename attribute is now marked as deprecated.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ jobs:
         - ["3.9",   "py39"]
         - ["3.10",  "py310"]
         - ["3.11",  "py311"]
+        - ["3.12",  "py312"]
+        - ["3.13.0-alpha - 3.13.99",  "py313"]
```
Comment (malthe) [inline .github/workflows/tests.yml:30]: Given the issues with Python 3.11 as demonstrated in #387, I guess supporting the unreleased Python 3.13 is a bit out of scope for this PR.

Comment (fschulze) [reply .github/workflows/tests.yml:30]: Adding nightly to the test matrix is meant to alert early, not saying we support it.

**Pull request merged**